pkg/model/iam.go

```diff
@@ -75,7 +75,15 @@ func (b *IAMModelBuilder) Build(c *fi.ModelBuilderContext) error {
 
 	// Generate IAM tasks for each shared role
 	for profileARN, igRole := range sharedProfileARNsToIGRole {
-		role, err := iam.BuildNodeRoleSubject(igRole)
+		lchPermissions := false
+		for _, ig := range b.InstanceGroups {
+			if ig.Spec.Role == igRole && ig.Spec.WarmPool != nil && ig.Spec.WarmPool.EnableLifecyleHook {
+				lchPermissions = true
```

Comment:
So if any IG sets `EnableLifecycleHook` then the permissions will be added to the nodes policy, allowing pods in any nodes group to abandon lifecycle hooks. It seems to me this setting should be cluster-level, not per-IG.